### ISSUE TITLE
Allow mapping custom extensions to language IDs

### ISSUE THREAD

**Issue by bencmbrook** [valentjn/languagetool-languageserver#122]
**Is your feature request related to a problem? Please describe.**

I'm trying to use LTeX with [`.mdx`](https://mdxjs.com/) files, which are almost entirely Markdown.

**Describe the solution you'd like**

While native support for MDX would be great (there are a few syntaxes that can throw the parser off, but pretty easy to ignore), it can also make sense to simply open a setting to map arbitrary file extensions to language IDs. In essence, I'd be enabling the "markdown" language for files ending with `.mdx`, as shown here:

https://github.com/valentjn/ltex-ls/blob/011a80b1e87797074740b74bf3cce6cdbdbd96f8/src/main/kotlin/org/bsplines/ltexls/tools/FileIo.kt#L120-L121

**Describe alternatives you've considered**

In VSCode, mapping `"files.associations": { "*.mdx": "markdown" }` successfully enables LTeX for `.mdx` files, but it's a global setting, so other extensions and syntax highlighters will also treat the file as Markdown.

**Additional context**

Another (imperfect) solution would be to simply add ` || fileName.endsWith(".mdx")` to the above snippet. There are also other common Markdown extensions that may be worth adding ([sindresorhus/markdown-extensions](https://github.com/sindresorhus/markdown-extensions/blob/main/markdown-extensions.json)). It's imperfect, because the parser doesn't recognize JSX snippets within the Markdown, and appears to treat them as empty string.

<img width="620" alt="Screen Shot 2021-11-13 at 1 43 11 PM" src="https://user-images.githubusercontent.com/7354176/141659876-0796fa81-1b8e-428b-b616-07081cf553ca.png">

**Comment by valentjn** [comment]
The Language Server Protocol (LSP) has the concept of [language identifiers](https://microsoft.github.io/language-server-protocol/specification#textDocumentItem), which are used by language servers to determine the type of documents. File extensions shouldn't be used as they are not reliable, they may be missing, and there may be clashes.

LT<sub>E</sub>X only uses file extensions (i.e., the source code quoted in the issue description) when running commands such as [`LTeX: Check All Documents in Workspace`](https://valentjn.github.io/ltex/vscode-ltex/commands.html#ltex-check-all-documents-in-workspace), because in this case, files are checked that are not open in the language client (editor) and therefore, no language identifier is available for them. For regular checking, file extensions are not used, only language IDs.

IMO, it would be confusing UX to have a setting that somehow supplements/overrides the editor choices of language IDs. Besides, I don't really see enough possible use cases besides MDX.

A better solution is to make the language client send a distinct language ID (`mdx` in this case), e.g., by installing [extensions that add language support for the specific language](https://marketplace.visualstudio.com/items?itemName=silvenon.mdx), and make the language server recognize the language ID.

Of course, most desirable would be adding full support for MDX to LT<sub>E</sub>X. Looking at examples, [MDX can be JavaScript with some Markdown in it](https://mdxjs.com/docs/what-is-mdx/#expressions), which would be pretty hard to parse and probably out-of-scope for LT<sub>E</sub>X. However, it's easily possible to check MDX as Markdown, if you're fine with all the false positives.

**Comment by bencmbrook** [comment]
Appreciate the thorough response and understand your reasoning. That global override can work well enough for my use case. 

Thanks—closing this out.
